examples/experimental/point-cloud-laz-vr/app.js

```diff
@@ -0,0 +1,245 @@
+const FILE_PATH = 'examples/point-cloud-laz/indoor.laz';
+
+function normalize(points) {
+  let xMin = Infinity;
+  let yMin = Infinity;
+  let zMin = Infinity;
+  let xMax = -Infinity;
+  let yMax = -Infinity;
+  let zMax = -Infinity;
+
+  for (let i = 0; i < points.length; i++) {
+    xMin = Math.min(xMin, points[i].position[0]);
+    yMin = Math.min(yMin, points[i].position[1]);
+    zMin = Math.min(zMin, points[i].position[2]);
+    xMax = Math.max(xMax, points[i].position[0]);
+    yMax = Math.max(yMax, points[i].position[1]);
+    zMax = Math.max(zMax, points[i].position[2]);
+  }
+
+  const scale = Math.max(...[xMax - xMin, yMax - yMin, zMax - zMin]);
+  const xMid = (xMin + xMax) / 2;
+  const yMid = (yMin + yMax) / 2;
+  const zMid = (zMin + zMax) / 2;
+
+  for (let i = 0; i < points.length; i++) {
+    points[i].position[0] = (points[i].position[0] - xMid) / scale;
+    points[i].position[1] = (points[i].position[1] - yMid) / scale;
+    points[i].position[2] = (points[i].position[2] - zMid) / scale;
+  }
+}
+
+class Example extends PureComponent {
+  constructor(props) {
+    super(props);
+
+    this._onViewportChange = this._onViewportChange.bind(this);
+    this._onInitialized = this._onInitialized.bind(this);
+    this._onResize = this._onResize.bind(this);
+    this._onUpdate = this._onUpdate.bind(this);
+
+    this.state = {
+      width: 0,
+      height: 0,
+      points: [],
+      progress: 0,
+      rotating: false,
+      vrDisplay: new EmulatedVRDisplay(),
+      emulatedPose: {
+        orientation: [0, 0, 0, 1],
+        position: [0, 0, 0]
+      }
+    };
+  }
+
+  componentWillMount() {
+    window.addEventListener('resize', this._onResize);
+    this._onResize();
+  }
+
+  componentDidMount() {
+    const {points} = this.state;
+
+    const skip = 50;
+    loadLazFile(`${DATA_REPO}/${FILE_PATH}`).then(rawData => {
+      parseLazData(rawData, skip, (decoder, progress) => {
+        for (let i = 0; i < decoder.pointsCount; i++) {
+          const {color, position} = decoder.getPoint(i);
+          points.push({color, position});
+        }
+
+        if (progress >= 1) {
+          normalize(points);
+        }
+
+        this.setState({points, progress});
+      });
+    });
+    this._initVRDisplay();
+    window.requestAnimationFrame(this._onUpdate);
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('resize', this._onResize);
+  }
+
+  _onResize() {
+    const size = {width: window.innerWidth, height: window.innerHeight};
+    this.setState(size);
+    this._onViewportChange();
+  }
+
+  _onInitialized(gl) {
+    setParameters(gl, {
+      clearColor: [0.07, 0.14, 0.19, 1],
+      blendFunc: [gl.SRC_ALPHA, gl.ONE_MINUS_SRC_ALPHA]
+    });
+  }
+
+  _onViewportChange(viewport) {
+    this.setState({
+      viewport: {...this.state.viewport, ...viewport}
+    });
+  }
+
+  _onUpdate() {
+    const {viewport} = this.state;
+    this.setState({
+      viewport: {
+        ...viewport,
```

Comment:
in my original example, I rotate the viewport for every animation frame.
we can remove the whole _onUpdate function if we don't need auto rotate and just the viewport generated from vrDisplay instead.